pkg/operator/edgefs/cluster/prepare/prepare.go

```diff
@@ -0,0 +1,261 @@
+	Version        string
+	serviceAccount string
+	placement      rookalpha.Placement
+	context        *clusterd.Context
+	resources      v1.ResourceRequirements
+	ownerRef       metav1.OwnerReference
+	ctx            context.Context
+}
+
+// New creates an instance of the prepare
+func New(
+	context *clusterd.Context, namespace, version string,
+	serviceAccount string,
+	placement rookalpha.Placement,
+	resources v1.ResourceRequirements,
+	ownerRef metav1.OwnerReference,
+) *Cluster {
+
+	if serviceAccount == "" {
+		// if the service account was not set, make a best effort with the example service account name since the default is unlikely to be sufficient.
+		serviceAccount = defaultServiceAccountName
+		logger.Infof("setting the prepare pods to use the service account name: %s", serviceAccount)
+	}
+
+	return &Cluster{
+		context:        context,
+		Namespace:      namespace,
+		serviceAccount: serviceAccount,
+		placement:      placement,
+		Version:        version,
+		resources:      resources,
+		ownerRef:       ownerRef,
+	}
+}
+
+// Start the prepare instance
+func (c *Cluster) Start(rookImage string, nodeName string) error {
+	logger.Infof("start running prepare pods")
+
+	// start the deployment
+	job := c.makeJob(appName, c.Namespace, rookImage, nodeName)
+	if _, err := c.context.Clientset.BatchV1().Jobs(c.Namespace).Create(job); err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return fmt.Errorf("failed to create %s job. %+v", appName, err)
+		}
+		logger.Infof("%s daemonset already exists", appName)
+	} else {
+		logger.Infof("%s job started", appName)
+	}
+
+	err := c.waitJob(job)
+	if err != nil {
+		logger.Warningf("Job %s waiting failed. %+v", job.ObjectMeta.Name, err)
+	}
+
+	err = c.deleteJob(job)
+	if err != nil {
+		logger.Warningf("Failed to delete job %s due. %+v", job.ObjectMeta.Name, err)
+	}
+
+	return nil
+}
+
+func (c *Cluster) makeJob(name, clusterName, rookImage string, nodeName string) *batch.Job {
+
+	volumes := []v1.Volume{
+		{
+			Name: etcVolumeName,
+			VolumeSource: v1.VolumeSource{
+				HostPath: &v1.HostPathVolumeSource{
+					Path: "/etc",
+				},
+			},
+		},
+	}
+
+	gracePeriod := int64(0)
+	podSpec := v1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   name,
+			Labels: c.getDaemonLabels(clusterName),
+		},
+		Spec: v1.PodSpec{
+			ServiceAccountName:            c.serviceAccount,
+			Containers:                    []v1.Container{c.prepareContainer(name, rookImage)},
+			RestartPolicy:                 v1.RestartPolicyOnFailure,
+			HostIPC:                       true,
+			HostNetwork:                   true,
+			TerminationGracePeriodSeconds: &gracePeriod,
+			NodeSelector: map[string]string{c.Namespace: "cluster",
+				"kubernetes.io/hostname": nodeName},
+			Volumes: volumes,
+		},
+	}
+	podSpec.Spec.DNSPolicy = v1.DNSClusterFirstWithHostNet
+	c.placement.ApplyToPodSpec(&podSpec.Spec)
+
+	ds := &batch.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name + "-" + nodeName,
```

Comment:
Please use the `TruncateNodeName` function here, as there are problems with Pods and other objects if their name is longer than 255 something chars including the cluster domain and other Service dns name parts.
https://github.com/rook/rook/blob/cd3d85e1e340ef09f7f530e641edd7ac0dc0005c/pkg/operator/k8sutil/k8sutil.go#L74-L87